### PR TITLE
Adding diesel to the cargetest suite

### PR DESCRIPTION
**Pull request by weiznich**
This was proposed in #79459 and #79560.
Diesel stresses the trait system implementation quite a lot and there
have been various regressions regarding that in the past.

### CODE REVIEW

File: src/tools/cargotest/main.rs [hunk 1]
```diff
@@ -48,6 +58,23 @@ const TEST_REPOS: &[Test] = &[
...
+    Test {
+        name: "diesel",
+        repo: "https://github.com/diesel-rs/diesel",
+        sha: "91493fe47175076f330ce5fc518f0196c0476f56",
+        lock: None,
+        packages: &[],
+        // Test the embeded sqlite variant of diesel
+        // This does not require any dependency to be present,
+        // sqlite will be compiled as part of the build process
+        features: Some(&["sqlite", "libsqlite3-sys/bundled"]),
```
Comment (jyn514) [inline src/tools/cargotest/main.rs:73]: How long does this take to compile?

**Comment by jyn514** [comment]
At some point it might also be interesting to add diesel to rustc-perf, but that doesn't necessarily have to happen now.

File: src/tools/cargotest/main.rs (hunk 1 above)
Comment (weiznich) [reply src/tools/cargotest/main.rs:73]: A clean build takes ~1min on my not really powerful laptop and maybe ~1 minutes to run all tests.

**Comment by weiznich** [comment]
> At some point it might also be interesting to add diesel to rustc-perf, but that doesn't necessarily have to happen now.

That was already proposed [here](https://github.com/rust-lang/rustc-perf/issues/232#issuecomment-391498299). I will try to have a look there as soon I've find some time. Diesel should definitively be an interesting workload there as it does spend most compile time inside of rustc and not in llvm. (At least that was the case last time I've looked at why does take that much time to compile).

File: src/tools/cargotest/main.rs (hunk 1 above)
Comment (jyn514) [reply src/tools/cargotest/main.rs:73]: Yeah, that sounds fine, thanks :)

**Comment by Mark-Simulacrum** [comment]
I don't think we should be expanding the set of crates in the cargotest suite until we have a T-compiler/T-infra FCP'd policy about how additions should get reviewed (e.g., who approves? how?). So I'm going to close this for now, but I'd love to see such a policy written; the first step would be to draft something reasonable and file an MCP to discuss that I think.

**Comment by weiznich** [comment]
@Mark-Simulacrum Who is the correct person to get in touch with to discuss creating such a policy? As already mentioned there where multiple diesel related issues in the last weeks + some more in the last year so this seems to be a good test case + diesel is also used as port of the official rust infrastructure (crates.io)

**Comment by jyn514** [comment]
@weiznich you can open an MCP here: https://github.com/rust-lang/compiler-team/issues/new/choose

but in general I think @Mark-Simulacrum 's point is that it *isn't* clear who to get in contact with and he's not comfortable making the decision unilaterally.

**Comment by pnkfelix** [comment]
Well then lets talk about it at the team meeting.

Nominating for discussion, in terms of the one-off situation of *this* particular test addition. (The more general policy can also be discussed, but I don't want that general discussion to distract from making a decision *here*.)

**Comment by pnkfelix** [comment]
reopened as PR #81507, so I'll nominated *that* for discussion.

**Pull request closed without merging**